### PR TITLE
En 10182/unit tests ap is

### PR DESCRIPTION
**Pull request by mariusmihaic**
Added tests for public APIs and wrappers:

- `PemFileReader`
- Account: `Address` + `Account`
-  Transaction: `Transaction` + `Signer`
- Wrappers: `libsodium` + `json`

Also, removed `PemFileReader` from **ih** namespace

### CODE REVIEW

File: tests/test_src/test_wrappers/test_wrappers.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    bytes const secretKey = wrapper::crypto::getSecretKey(seed);
+
+    std::string const msg = "{\"nonce\":0,\"value\":\"0\",\"receiver\":\"erd1cux02zersde0l7hhklzhywcxk4u9n4py5tdxyx7vrvhnza2r4gmq4vw35r\",\"sender\":\"erd1l453hd0gt5gzdp7czpuall8ggt2dcv5zwmfdf3sd3lguxseux2fsmsgldz\",\"gasPrice\":1000000000,\"gasLimit\":50000,\"data\":\"Zm9v\",\"chainID\":\"1\",\"version\":1}";
+    std::string const signature = wrapper::crypto::getSignature(secretKey, msg);
+    std::string const expectedSignature = "b5fddb8c16fa7f6123cb32edc854f1e760a3eb62c6dc420b5a4c0473c58befd45b621b31a448c5b59e21428f2bc128c80d0ee1caa4f2bf05a12be857ad451b00";
+
+    EXPECT_EQ(signature.size(), SIGNATURE_LENGTH);
+    EXPECT_EQ(util::stringToHex(signature), expectedSignature);
+
+}
```
Comment (miiu96) [inline tests/test_src/test_wrappers/test_wrappers.cpp]: add empty line

Comment (mariusmihaic) [reply tests/test_src/test_wrappers/test_wrappers.cpp]: Fixed it !

File: tests/test_src/CMakeLists.txt [hunk 2]
```diff
@@ -1,2 +1,5 @@
 add_subdirectory(test_utils)
-
+add_subdirectory(test_account)
```
Comment (AdoAdoAdo) [inline tests/test_src/CMakeLists.txt:2]: 👍  for splitting into separate subdirectories

**Review by AdoAdoAdo — APPROVED** [review]

**Review by miiu96 — APPROVED** [review]

File: tests/test_src/CMakeLists.txt (hunk 2 above)
Comment (mariusmihaic) [reply tests/test_src/CMakeLists.txt:2]: Thanks ! :smile:

**Pull request merged**